cmake_modules/FindLIBMEMCACHED.cmake

```diff
@@ -21,47 +21,126 @@
 #  LIBMEMCACHED_INCLUDE_DIR - the libmemcached include directory(s)
 #  LIBMEMCACHED_LIBRARIES - The libraries needed to use libmemcached
 
+#  If the memcached libraries are found on the system, we assume they exist natively and dependencies
+#  can be handled through package management.  If the libraries are not found, and if
+#  MEMCACHED_USE_EXTERNAL_LIBRARY is ON, we will fetch, build, and include a copy of the neccessary
+#  Libraries.
+
+
 IF (NOT LIBMEMCACHED_FOUND)
-  IF (WIN32)
-    SET (libmemcached_lib "libmemcached")
-    SET (libmemcachedUtil_lib "libmemcachedutil")
-  ELSE()
-    SET (libmemcached_lib "memcached")
-    SET (libmemcachedUtil_lib "memcachedutil")
-  ENDIF()
-
-  FIND_PATH(LIBMEMCACHED_INCLUDE_DIR libmemcached/memcached.hpp PATHS /usr/include /usr/share/include /usr/local/include PATH_SUFFIXES libmemcached)
-
-  FIND_LIBRARY (LIBMEMCACHEDCORE_LIBRARY NAMES ${libmemcached_lib} PATHS /usr/lib usr/lib/libmemcached /usr/share /usr/lib64 /usr/local/lib /usr/local/lib64)
-  FIND_LIBRARY (LIBMEMCACHEDUTIL_LIBRARY NAMES ${libmemcachedUtil_lib} PATHS /usr/lib /usr/share /usr/lib64 /usr/local/lib /usr/local/lib64)
-
-  #IF (LIBMEMCACHED_LIBRARY STREQUAL "LIBMEMCACHED_LIBRARY-NOTFOUND")
-  #  SET (LIBMEMCACHEDCORE_LIBRARY "")    # Newer versions of libmemcached are header-only, with no associated library.
-  #ENDIF()
-
-  SET (LIBMEMCACHED_LIBRARIES ${LIBMEMCACHEDCORE_LIBRARY} ${LIBMEMCACHEDUTIL_LIBRARY})
-
-  IF(LIBMEMCACHED_INCLUDE_DIR)
-    FILE (STRINGS "${LIBMEMCACHED_INCLUDE_DIR}/libmemcached-1.0/configure.h" version REGEX "#define LIBMEMCACHED_VERSION_STRING")
-    STRING(REGEX REPLACE "#define LIBMEMCACHED_VERSION_STRING " "" version "${version}")
-    STRING(REGEX REPLACE "\"" "" version "${version}")
-    SET (LIBMEMCACHED_VERSION_STRING ${version})
-    IF ("${LIBMEMCACHED_VERSION_STRING}" VERSION_EQUAL "${LIBMEMCACHED_FIND_VERSION}" OR "${LIBMEMCACHED_VERSION_STRING}" VERSION_GREATER "${LIBMEMCACHED_FIND_VERSION}")
-      SET (LIBMEMCACHED_VERSION_OK 1)
-      SET (MSG "${DEFAULT_MSG}")
+    option(MEMCACHED_USE_EXTERNAL_LIBRARY "Pull and build source from external location" OFF)
```

Comment:
I did it this way because if we have the minimum required version available by the system, we should always try to use that instead of packaging a third party library in with our plugin package.  I was assuming this would be useful to set for on if we're trying to build on a system, and we find that there isn't a sufficient version installed (i.e. centos 5)